### PR TITLE
SDIT-2192 Add Splash Condition to Splash Screen and add /blocked endpoint

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/SplashScreenResource.kt [hunk 1]
```diff
@@ -62,20 +62,83 @@ class SplashScreenResource(private val service: SplashScreenService) {
...
     @Schema(description = "The name of the screen (module)", example = "OIDINCRS")
     @PathVariable moduleName: String,
-  ): SplashScreenDto = service.getSplashScreens(moduleName)
+  ): SplashScreenDto = service.getSplashScreen(moduleName)
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_SCREEN_ACCESS')")
+  @GetMapping("/splash-screens/{moduleName}/blocked")
+  @Operation(
+    summary = "Retrieve a list of blocked prison ids for the screen",
+    description = "Retrieves a list of blocked prison ids for the screen (module) name or **ALL* if all prisons, or an empty list if there are none blocked. Requires role NOMIS_SCREEN_ACCESS",
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/SplashScreenResource.kt:71]: V. minor type on `**ALL*`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/SplashScreenResource.kt [hunk 2]
```diff
@@ -62,20 +62,83 @@ class SplashScreenResource(private val service: SplashScreenService) {
...
+  @Schema(description = "The type of access - YES, NO, COND")
+  val accessBlockedType: CodeDescription,
 
   @Schema(description = "The text shown when a screen is accessible but will shortly be turned off", example = "This screen will be turned off next month.")
   val warningText: String?,
 
   @Schema(description = "The text shown when a screen is blocked", example = "This screen is no longer accessible, use DPS.")
   val blockedText: String?,
+
+  @Schema(description = "The prison code", example = "BXI")
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/SplashScreenResource.kt:142]: Minor but I don't think this matches the field type?

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/SplashScreenResource.kt (hunk 1 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/SplashScreenResource.kt:71]: oopsie - will add in - I am doing another PR so will include in that

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/SplashScreenResource.kt (hunk 2 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/SplashScreenResource.kt:142]: ah yes - cut and paste - will also fix in next PR

**Pull request merged**